design/src/rfcs/rfc0006_crates_io_publishing.md

```diff
@@ -0,0 +1,269 @@
+for example. The `patch` version is allowed to be different between service crates.
+
+Smithy runtime crates will maintain their own versioning, and their version numbers are independent of each other.
+For example, the latest `smithy-types` could be at `0.3` while `smithy-json` is at `0.2`. The `aws-config` crate
+is an exception to this rule since customers depend on it directly. It should maintain the same `major.minor` as
+the AWS SDK crates.
+
+The `pre` version tag will be `alpha` during the Rust SDK alpha, `preview` during dev preview, and removed once the
+SDK is generally available.
+
+### Version Bumping
+
+In general, model changes will be frequent (daily) while changes to runtime crates and code generation
+will be infrequent (weekly to monthly). Model changes are supposed to _always_ be backwards compatible, so it
+makes sense to apply these to `patch` revisions as much as possible. That way, if a model change was made only
+to S3, then only the S3 crate would be changed rather than every AWS SDK crate, of which there are more than 260.
+
+AWS SDK crates will follow this process for version bumps:
+
+1. Bump the `minor` version in tandem with _all_ other SDK crates and `aws-config` if the smithy-rs version used to
+   generate the SDK changed. This ensures changes to runtime crates and/or code generation logic will result in
+   a `minor` version bump.
+2. Bump the `patch` version if the smithy-rs version remained the same _AND_ if there are code changes to
+   the SDK crate. This allows backwards compatible model changes to apply as patch revisions.
+
+The version bump requirements for runtime crates are flexible and deferred to the short-term and long-term proposals.
+
+### Yanking
+
+Mistakes will inevitably be made, and a mechanism is needed to yank packages while keeping the latest version
+of the SDK successfully consumable from crates.io. To keep this simple, the entire published batch of crates
+will be yanked if any crate in that batch needs to be yanked. For example, if 260 crates were published in a batch,
+and it turns out there's a problem that requires yanking one of them, then all 260 will be yanked. Attempting to do
+partial yanking will require a lot of effort and be difficult to get right. Yanking should be a last resort.
+
+Concrete Scenarios
+------------------
+
+The following are example scenarios that may occur that the publishing implementation must handle.
+
+1. **AWS models were updated**: This results in `patch` version bumps for all AWS SDK crates _that had changes_.
+2. **Bug fixed in runtime crate**: Runtime crate is `minor` version bumped. All runtime crates that depend on it are
+   also `minor` version bumped, and all of the AWS SDK crates are `minor` version bumped.
+3. **Bug fixed in codegen**: All AWS SDK crates are `minor` version bumped.
+4. **CVE discovered in external dependency**: If the external dependency is fixed in a `patch` revision where it will
```

Comment:
depending on severity it seems like this may necessitate yanking as well?

Reply:
I think this design allows for yanking old vulnerable versions if we decide to do it since we could retrieve old version manifest files out of git history and generate yank commands from them.